tests/compat.py

```diff
@@ -0,0 +1,117 @@
+
+curr_dir = os.path.dirname(os.path.abspath(__file__))
+curr_dir = os.path.dirname(curr_dir)
+build_dir = os.path.join(curr_dir, 'build')
+cpp_test_filename = 'taichi_cpp_tests'
+capi_test_filename = 'taichi_c_api_tests'
+cpp_tests_path = os.path.join(build_dir, capi_test_filename)
+c_api_tests_path = os.path.join(build_dir, cpp_test_filename)
+
+# aot_copy_list = [c_api_tests+'--gtest_filter=CapiTest.TestCompat*']
+# graph_aot_test_list = [c_api_tests+' --gtest_filter=CapiTest.TestCompatLoadAOT']
+# aot_module_test_list = [c_api_tests+' --gtest_filter=CapiTest.TestBehaviorLoadAOTModuleVulkan']
+# tcm_test_list = [c_api_tests+' --gtest_filter=CapiTest.*TcmAotModule']
+# kernel_aot_test1_list = [c_api_tests+' --gtest_filter=CapiTest.AotTestVulkanKernel', c_api_tests+' --gtest_filter=CapiTest.DryRunVulkanAotModule']
+# dense_field_aot_test_list = [cpp_tests+' --gtest_filter=GfxAotTest.VulkanDenseField']
+# kernel_aot_test2_list = [cpp_tests+' --gtest_filter=GfxAotTest.VulkanKernelTest2',cpp_tests+' --gtest_filter=CGraphAotTest.VulkanRunCGraph2']
+# mpm88_graph_aot_list = [c_api_tests+' --gtest_filter=CapiTest.Mpm88TestVulkan']
+# sph_aot_list = [c_api_tests+' --gtest_filter=CapiTest.SphTestVulkan']
+# tcm_test_list = [c_api_tests+' --gtest_filter=CapiTest.TestLoadTcmAotModule',c_api_tests+' --gtest_filter=CapiTest.TestCreateTcmAotModule']
+# texture_aot_test_list = [c_api_tests+' --gtest_filter=CapiTest.GraphTestVulkanTextureGraph']
+
+# run_dict = {'aotcopy':aot_copy_list,'aot_module_test':aot_module_test_list,'dense_field_aot_test':dense_field_aot_test_list,
+# 'graph_aot_test':graph_aot_test_list,'kernel_aot_test1':kernel_aot_test1_list,'kernel_aot_test2':kernel_aot_test2_list,
+# 'mpm88_grpah_aot':mpm88_graph_aot_list,'sph_aot':sph_aot_list,'tcm_test': tcm_test_list,'texture_aot_test': texture_aot_test_list}
+
+run_dict = {}
+
+
+def init_dict(run_dict, aot_files):
+    curr_dir = os.path.dirname(os.path.abspath(__file__))
+    test_config_path = os.path.join(curr_dir, 'test_config.json')
+    with open(test_config_path, 'r') as f:
+        test_config = json.loads(f.read())
+
+    assert ("aot_test_cases" in test_config.keys())
+    assert ("capi_aot_test_cases" in test_config.keys())
+
+    for x in aot_files:
+        path_name = pathlib.Path(x).name[:-3]
+        run_dict[path_name] = []
+    for cpp_test_name, value in test_config["aot_test_cases"].items():
+        if value[1] != "--arch=vulkan": continue
+        test_command = []
+        test_command.append(cpp_tests_path)
+        test_command.append(f"--gtest_filter={cpp_test_name}")
+        run_dict[value[0][3][:-3]].append(test_command)
+
+    for cpp_test_name, value in test_config["capi_aot_test_cases"].items():
+        if value[1] != "--arch=vulkan": continue
+        test_command = []
+        test_command.append(cpp_tests_path)
+        test_command.append(f"--gtest_filter={cpp_test_name}")
+        run_dict[value[0][3][:-3]].append(test_command)
+
+
+def generate():
```

Comment:
Make them separate scripts with more readable names like `generate_test_modules.py` and `run_c_api_tests.py`. And for the better you wanna reuse this code from `run_tests.py`.